runtime/drivers/git/git.go

```diff
@@ -0,0 +1,103 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	gogit "github.com/go-git/go-git/v5"
+	"github.com/rilldata/rill/runtime/drivers"
+	"go.uber.org/zap"
+)
+
+func init() {
+	drivers.Register("git", driver{})
+}
+
+type driver struct{}
+
+func (d driver) Open(dsn string, logger *zap.Logger) (drivers.Connection, error) {
+	c := &connection{root: dsn}
+	if err := c.checkRoot(); err != nil {
+		err = os.RemoveAll("tempdir")
+		if err != nil {
+			return nil, err
+		}
+
+		split := strings.Split(dsn, "|")
+		if len(split) > 1 {
+			dsn = split[1]
+		}
+
+		_, err := gogit.PlainClone("tempdir", false, &gogit.CloneOptions{
+			URL:      dsn,
+			Progress: os.Stdout,
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+	return c, nil
+}
+
+type connection struct {
+	root string
+}
```

Comment:
It should use `os.MkdirTemp("", "rill_git")` to create a temp dir for storing the repo. Then in `Close()`, it should delete the temp dir.